### PR TITLE
fix(sandbox): Fix docker root login and set default to run-as-devin

### LINKED ISSUE #887: Running as root is not supported at the moment.
The Traceback is like: 
 
image

Here is my config:
image

### PR DESCRIPTION
**Pull request by xingyaoww**
Fix #887.

As a side note: I think it is probably better to run as Devin by default since it won't mess up the file permission of files written to the mounted directory. We already make `opendevin` user in the `sudoer`'s group anyway, so theoretically it can still perform root commands (e.g., `sudo apt-get install`) when absolutely needed.

### CODE REVIEW

**Comment by yufansong** [comment]
Do a hotfix here #895 to at least make opendevin can work first.

**Comment by yufansong** [comment]
When I find your preivous PR, it is already merged. I trid it locally find there will meet some docker network problem. I also tried this PR, also `ssh` will meet problem locally.
```
                
02:21:56 - opendevin:INFO: sandbox.py:118 - Using workspace directory: /Users/yufansong/code/test/OpenDevin/workspace
02:21:57 - opendevin:INFO: sandbox.py:300 - Container stopped
02:21:57 - opendevin:INFO: sandbox.py:322 - Container started
02:21:58 - opendevin:INFO: sandbox.py:338 - waiting for container to start: 1, container status: running
02:21:58 - opendevin:ERROR: sandbox.py:374 - Failed to start Docker container: Could not establish connection to host
Traceback (most recent call last):
  File "/Users/yufansong/code/test/OpenDevin/opendevin/sandbox/sandbox.py", line 370, in <module>
    docker_interactive = DockerInteractive(
  File "/Users/yufansong/code/test/OpenDevin/opendevin/sandbox/sandbox.py", line 138, in __init__
    self.start_ssh_session()
  File "/Users/yufansong/code/test/OpenDevin/opendevin/sandbox/sandbox.py", line 181, in start_ssh_session
    self.ssh.login(hostname, username, self._ssh_password, port=2222)
  File "/Users/yufansong/miniforge3/envs/opendevin/lib/python3.10/site-packages/pexpect/pxssh.py", line 424, in login
    raise ExceptionPxssh('Could not establish connection to host')
pexpect.pxssh.ExceptionPxssh: Could not establish connection to host
```
I slightly changed the network setting to make it work. Want to make it back to work at first. But feel free to revert it if we find it is my own problem. I think we need to be cautious when introduce large change.

**Comment by yufansong** [comment]
Seem not only my own problem: https://github.com/OpenDevin/OpenDevin/pull/895#issuecomment-2042300629

**Comment by xingyaoww** [comment]
@yufansong, thanks for the quick PR. I think the issue is probably because of the Allow Origin 127.0.0.1, which may cause issues for different setups. I don't think changing `network_mode='host',` to the port mapping of `2222` will have a huge impact? It may further stop the user from accessing web servers started by opendevin inside the sandbox.

**Comment by yufansong** [comment]
> @yufansong, thanks for the quick PR. I think the issue is probably because of the Allow Origin 127.0.0.1, which may cause issues for different setups. I don't think changing `network_mode='host',` to the port mapping of `2222` will have a huge impact? It may further stop the user from accessing web servers started by opendevin inside the sandbox.

Yeah, agree.

**Comment by JustinLin610** [comment]
Everything works now for me. Somehow ugly for the 2222 but it seems to be a reasonable solution for now.

File: opendevin/sandbox/sandbox.py [hunk 1]
```diff
@@ -292,12 +326,25 @@ def restart_docker_container(self):
...
+                     'Server started by OpenDevin will not be accessible from the host machine at the moment. '
+                     'See https://github.com/OpenDevin/OpenDevin/issues/897 for more information.'
+                     )
+                )
+
             # start the container
             self.container = docker_client.containers.run(
                 self.container_image,
-                # only allow connections from localhost
-                command="/usr/sbin/sshd -D -p 2222 -o 'ListenAddress=0.0.0.0'",
```
Comment (SmartManoj) [inline opendevin/sandbox/sandbox.py:299]: Why not -o 'ListenAddress=127.0.0.1' -o 'ListenAddress=172.17.0.1'?

**Pull request merged**